rbx_binary/src/text_deserializer.rs

```diff
@@ -649,6 +690,25 @@ pub enum DecodedChunk {
     },
 }
 
+fn shared_string_serializer<S>(values: &[SharedString], serializer: S) -> Result<S::Ok, S::Error>
+where
+    S: Serializer,
+{
+    let mut processed = Vec::with_capacity(values.len());
+
+    for value in values {
+        let hash = value.hash();
+        let mut hash_hex = String::with_capacity(hash.as_bytes().len() * 2);
+
+        for byte in hash.as_bytes() {
+            write!(hash_hex, "{:02x}", byte).unwrap();
+        }
+        processed.push(format!("[{} bytes] {}", value.data().len(), hash_hex))
+    }
+
+    serializer.collect_seq(processed)
```

Comment:
Using `collect_seq` here when we aren't generating an iterator is pretty weird. You should use [`serialize_seq`](https://docs.rs/serde/1.0.117/serde/trait.Serializer.html#tymethod.serialize_seq) instead, then call the methods on its result. This lets us avoid needing to allocate a `Vec` here needlessly.

We can also reuse the `String` between loop iterations. Allocate it at the beginning of this function, `clear()` it at the beginning of the loop, and then write into it and use `serialize_element` on it.

Reply:
Another question. Can we serialize these values the same way that we serialize them for `DomViewer`? We should be able to make up a little struct that derives `Serialize` to use here:

```rust
#[derive(Serialize)]
struct SerializedSharedString<'a> {
    len: usize,
    hash: &'a str,
}
```

Then we can create one, serialize it into the sequence, and happy re-use the `String` it referred to for the next value.